Source/Core/Core/HW/Sram.cpp

```diff
@@ -58,9 +58,9 @@ void InitSRAM()
   }
 }
 
-void SetCardFlashID(u8* buffer, u8 card_index)
+void SetCardFlashID(const u8* buffer, u8 card_index)
 {
-  u64 rand = Common::swap64(*(u64*)&(buffer[12]));
+  u64 rand = Common::swap64(&buffer[12]);
```

Comment:
It's correct, because the `swap64` that takes a `const u8*` does the same thing. https://github.com/dolphin-emu/dolphin/blob/7139b6bae6301f667a9ebff2ec7f7dc1820bbf2c/Source/Core/Common/CommonFuncs.h#L201